Release 3.18.0

jennyf19 opened the pull request:
Release 3.17.4 w/GDPR & sovereign cloud

rayluo committed:
New instance discovery implementation

rayluo committed:
Plug new InstanceDiscovery into Authenticator

rayluo committed:
Remove obsolete AuthenticatorTemplate and its list

rayluo committed:
Add unit tests for the new Instance Discovery

rayluo committed:
Address sonarqube feedbacks

rayluo committed:
Adding line break after some closing curly brackets

rayluo committed:
Adding .ConfigureAwait(false) to all await lines

rayluo committed:
Adjust inline comments

rayluo committed:
Avoid contextual keyword alias

rayluo committed:
Use string interpolation

rayluo committed:
More explanation on the exception handling

rayluo committed:
Do not store instance meta in Authenticator

rayluo committed:
Rename test case names to follow func_when_should() convention

rayluo committed:
Sync up with dev branch and have to adjust some extra test cases

rayluo committed:
Use preferred network to talk to endpoints

rayluo committed:
Rollback modifications on sovereign cloud test cases

rayluo committed:
Also assert the correct host in test cases

rayluo committed:
Merge branch 'rayluo/lwn-mso-migration-2' into dev

Commit:
add alpha for all branches except master

SomkaPe committed:
fixes for Android and IOS platforms (#823)

* xamarin ios app fixes

* fix for null pointer exception on Android CanSwitchToBroker call, fix for applying SupportToAddUserThroughBroker to company portal broker

jennyf19 committed:
Ensure prompt behavior for each value of prompt behavior (#818)

* Ensure prompt behavior for each value of prompt behavior
- test calling promptBehavior.Auto with AT in cache
- Create class for PromptBehavior tests and move existing prompt tests from AdalDotNetTests

* Clean-up tests

* Moving prompt tests to integration test folder

trwalke committed:
Created a negative test for acquiring device code. #804 (#819)

* Created a negative test for acquiring device code. #804

* Removed specific error message, added code to properly handle expected exceptions. #804

trwalke committed:
Created one negative test for acquiring token using device code where… (#825)

* Created one negative test for acquiring token using device code where the polling times out.

Resolved issue where ADAL did not return the STS response after a device code has expired.

#804

* Bug Fix #804

* Added comment about code change

#804

trwalke committed:
Created a test case where a valid access token exists in the cache fo… (#827)

* Created a test case where a valid access token exists in the cache for AcquireTokenAsync.

* Cleared token cache for failing taken test.

* Requested changes made to clean up code.

* Clean up.

jennyf19 committed:
Jennyf/remove ws trust test (#829)

* Ensure prompt behavior for each value of prompt behavior
- test calling promptBehavior.Auto with AT in cache
- Create class for PromptBehavior tests and move existing prompt tests from AdalDotNetTests

* Clean-up tests

* Moving prompt tests to integration test folder

* Remove WsTrust test

Commit:
refactored test projects to remove dependencies per #830 (#831)

* refactored test projects to remove dependencies per #830

* fix namespace

trwalke committed:
Created a test case where cache is empty. ADAL should return error. (#833)

* Created a test case where cache is empty. ADAL should return error.

Created a test case where a expired access token and a invalid refresh token exist in cache.

* Requested changes made.

Commit:
Kpanwar/readme changelog 814 correct (#837)

* fixes for Android and IOS platforms (#821)

* xamarin ios app fixes

* fix for null pointer exception on Android CanSwitchToBroker call, fix for applying SupportToAddUserThroughBroker to company portal broker

* changelog for 3.17.0 and README

rayluo committed:
TokenCache read adjustments

rayluo committed:
Skip instance discovery when using adfs authority (#838)

trwalke committed:
Refactored existing tests to integration project #840 (#841)

* Refactored existing tests to integration project #840

* Renamed file #840

* Renamed test #840

* Added positive test for AcquireToken with an empty cache.
#840

Commit:
Only keep public classes in Microsoft.IdentityModel.Clients.ActiveDirectory (#843)

* namespace fix and refactoring

* update namespace.

* fix automation app and winrt solution

* address sonarqube comments

* not use shorthands

* fix setters

jennyf19 committed:
Jennyf/prompt behavior test fix (#846)

* Fixing bug in PromptBehavior.Auto test case

* Addressing Appveyor failure

Commit:
update version for next release

rayluo committed:
Adding test cases to prove the previous 2-liners is sufficient

rayluo committed:
TokenCache write adjustments

rayluo committed:
Add .ConfigureAwait(false)

trwalke committed:
Ensure Managed User Credentials integration tests exist in ADAL.NET PR (#849)

* Test case with empty cache where network call is made to user realm endpoint and then to token endpoint to get token from AAD

Test case where valid access token already exists in cache. this should result in cache token returned back.

Test case where valid access token for user1 already exists in cache. API is called for user2 and in this case network call is made to user realm endpoint and then to token endpoint to get token from AAD. Now the cache should have 2 entries.

Test case with expired access token and valid refresh token in cache. This should result in refresh token being used to get new AT instead of user creds.

Test case where user realm discovery fails because user does not exist in AAD.

Test case where user realm discovery cannot determine the user type

#840
VSTS 292990

* Addressed requested changes

#840

* Added License.
clean up

#840

* Ensured that the HttpMessageHandlerFactory que was clear

#840

rayluo committed:
Merge branch 'rayluo/lwn-mso-migration-tokencache-2' into dev

jennyf19 committed:
Jennyf/user credential tests (#853)

* Create class for Federated user credential tests
- test for empty cache and username/password flow

* - In MockHttpMessageHandler, add in constructor and constructor plus overload to specify url endpoint
- Add Acquire Token w/empty cache for Federated user test
- Change output property for TestMex.xml and WsTrustResponse.xml to Always
- Update TextMex.xml from network call
- Update WsTrustResponse.xml from network call

* Addressing VSTS issues...removing unnecssary references

* Removing unnecssary references to xamarin

* moving and removing files

* Add WsTrustResponse xml for WsTrust13
Make changes to NonInteractiveTests to use correct xml doc

* change test name

jennyf19 committed:
Jennyf/user credential tests (#856)

* Create class for Federated user credential tests
- test for empty cache and username/password flow

* - In MockHttpMessageHandler, add in constructor and constructor plus overload to specify url endpoint
- Add Acquire Token w/empty cache for Federated user test
- Change output property for TestMex.xml and WsTrustResponse.xml to Always
- Update TextMex.xml from network call
- Update WsTrustResponse.xml from network call

* Addressing VSTS issues...removing unnecssary references

* Removing unnecssary references to xamarin

* moving and removing files

* Add WsTrustResponse xml for WsTrust13
Make changes to NonInteractiveTests to use correct xml doc

* change test name

* Change cloudinstancename in mocks

jennyf19 committed:
Jennyf/user credential tests (#857)

#852 
* Create class for Federated user credential tests
- test for empty cache and username/password flow

* - In MockHttpMessageHandler, add in constructor and constructor plus overload to specify url endpoint
- Add Acquire Token w/empty cache for Federated user test
- Change output property for TestMex.xml and WsTrustResponse.xml to Always
- Update TextMex.xml from network call
- Update WsTrustResponse.xml from network call

* Addressing VSTS issues...removing unnecssary references

* Removing unnecssary references to xamarin

* moving and removing files

* Add WsTrustResponse xml for WsTrust13
Make changes to NonInteractiveTests to use correct xml doc

* change test name

* Change cloudinstancename in mocks

* Add mex endpoint fails to resolve test

* Add federated user credential tests:
- Provide username only of federated user test
- No password provided and mex does not return integrated auth endpoint test
- Test for acquire token w/valid token in cache
- Mex endpoint fails to resolve and results in 404 test

* Addressing PR comments

rayluo committed:
String interpolation is not fully supported in 4.5 (#860)

This is exactly the same issue mentioned in this StackOverflow question:
https://stackoverflow.com/questions/32076823/string-interpolation-in-visual-studio-2015-and-iformatprovider-ca1305

Since we are not targeting framework 4.6 yet, we will have to switch back
to the old school string.Format(...)

rayluo committed:
Merge branch 'rayluo/lwn-mso-migration-preferred-network' into dev

jennyf19 committed:
Jennyf/user credential tests (#861)

* Create class for Federated user credential tests
- test for empty cache and username/password flow

* - In MockHttpMessageHandler, add in constructor and constructor plus overload to specify url endpoint
- Add Acquire Token w/empty cache for Federated user test
- Change output property for TestMex.xml and WsTrustResponse.xml to Always
- Update TextMex.xml from network call
- Update WsTrustResponse.xml from network call

* Addressing VSTS issues...removing unnecssary references

* Removing unnecssary references to xamarin

* moving and removing files

* Add WsTrustResponse xml for WsTrust13
Make changes to NonInteractiveTests to use correct xml doc

* change test name

* Change cloudinstancename in mocks

* Add mex endpoint fails to resolve test

* Add federated user credential tests:
- Provide username only of federated user test
- No password provided and mex does not return integrated auth endpoint test
- Test for acquire token w/valid token in cache
- Mex endpoint fails to resolve and results in 404 test

* Addressing PR comments

* Adding Federated user test coverage
- test case w/no password and mex does not return integrated auth endpoint
- test case w/password provided and mex does not return username/password endpoint
- test case w/expired access token and valid refresh token in cache.

* fix url

* Validate cache entry is updated w/new AT
Check all mocks are consumed
Update to AT silent test

jennyf19 committed:
Minor edits to managed and federated user tests (#869)

- fix comments
- add token cache count check

danieldobalian committed:
Updated comment with possible x-client-skus (#864)

trwalke committed:
Added URL requirement for MockHttpMessageHandler #847 (#866)

* Added URL requirement for MockHttpMessageHandler #847

# Conflicts:
#	tests/Test.ADAL.NET.Unit/InstanceDiscoveryTests.cs

* Clean up #847

* Clean up #847

* clean up missing variable.

* Removed unnecessary binding to NSubstitute #847
Clean up

* Refactored Url functions #847

* Refactored url functions #847

* Renamed endpoint functions.
Refactored "userrealm".
#847

rayluo committed:
Authority Migration implementation includes tenant info now (#876)

* Refactor to include tenant in Authority Migration

* Change old test cases to use lower case

These 2 mixed-case tests came from the very original commit of this code base.
However hosts are not supposed to be case sensitive.

These 2 test cases tend to fail because:

1. Our pre-existing token cache implementation happens to use the
dev-provided host as-is to build a cache key, without normalizing the host
part first. (We should probably normalize them, but we did not).

2. The current and probably any new implementation will use `Uri uri`
rather than `string uri`, which will automatically normalize `uri.Host`
into lower case;
(https://blogs.msdn.microsoft.com/brada/2004/01/12/design-guideline-update-uri-vs-string/)

3. The mismatch of the above 2 facts gives us the PITA.

* Use a real response in the default test case

It does not change the test cases behaviors though. And it should not.

* Add null check for public methods

Commit:
merge Release/3.17.1 branch back into dev (#886)

* merge dev to master for 3.16.0 (#749)

* Implement PromptBehavior to iOS/Android

Implement PromptBehavior to iOS/Android.
Add PromptBehavior Property to PlatformParameters.
Implement GetCacheLoadPolicy() and
AddPromptBehaviorQueryParamete() in PlatformInformation

* Remove PromptBehavior.Never from iOS/Android platforms

We don't support PromptBehavior.Never on the iOS and Android platforms.
Move the PromptBehavior.cs file back out of the common PCL project and
use platform specific (WinRT/Desktop & iOS/Android) versions.

* Tidy up some code.

* Fix bad merge

* Suppress various FxCop issues (track VSTS 265363)

* Added targets file to work around VSTS SDL FxCop issue

* Modified FxCop fix target to accept a root folder path

* Updated description in FxCop workaround targets

* Added missing null checks to CryptographyHelper.cs implementations.

* Add GetAuthRequestUrlAsync method without claims overload
Add unit tests for GetAuthReqeustUrlAsync without claims overload

* Url encode claims for iOS with Broker

* prepare for 3.15.1

prepare for 3.15.1

* bump to 3.16.0

* Update changelog.txt

* Fix to use RSACng when RSA CSP is not resolved (#779)

* fix net47 crypto issue

* update build script to accomodate alpha builds

* changelog for 3.16.1 (#785)

* fixes for Android and IOS platforms (#821)

* xamarin ios app fixes

* fix for null pointer exception on Android CanSwitchToBroker call, fix for applying SupportToAddUserThroughBroker to company portal broker

* changelog for 3.17.0 and README (#836)

* fix bad merge

* fix promptbehavior enum and not use extension methods

* rename file

* update to 3.17.1

* create new test certificate

* fix doc

* add obsolete tags

* move AdalError to separate file

rayluo committed:
Refactor previous test cases to reduce dependency

Commit:
base checkin

Commit:
cleanup csproj file

rayluo committed:
Reset InstanceDiscovery during integration tests

As required by Kanishk

Commit:
add typeforward for all relevant projects

Commit:
generate doc xml and layout platform dlls

Commit:
keep the same versioning

Commit:
fix android

rayluo committed:
Merge branch 'rayluo/refactor-old-test-cases' into dev

Commit:
add 1.1

Commit:
update layout

Commit:
use platform specific output only

Commit:
fix versioning

Commit:
fix winRT versioning

trwalke committed:
Added certificate as an x5c claim to header of JWT request tokens. (#883)

* Added certificate as an x5c claim to header of JWT request tokens.

* Update IClientAssertionCertificate.cs

* Reworked code to support developer implementations of IClientAssertionCertificate with an X509 claim.

Added unit test to check for c5x claim in JWT request header.

* Clean up.

* Added test Initialization.

* Added test for developer implemented client assertion with X509

* Removed x5c claim from developer implemented IClientAssertionCertificate.
clean Up.

* Made certificate property auto-implement.

* changed detection of empty x5c claim in JWT header.

* Refactored test locations.
JsonWebToken client credential flow tests now target tenant specific endpoints.
Clean Up.

* Fixed mismatched URL encoding.
Changed x5c encoding to use raw certificate data.

Commit:
include file for compilation

Commit:
Update changelog.txt

Commit:
merge hotfix into dev

Commit:
update dev for changes in master

jennyf19 committed:
Jennyf/reenable sovereign cloud (#884)

* Minor edits to managed and federated user tests
- fix comments
- add token cache count check

* Reenable Sovereign Cloud
- return task for PostTokenRequest
- Update Authority in ATHandlerBase
- UpdateAuthority in ATInteractiveHandler
- add authority to tokenReponse
- add authority to TokenResponse in android broker
- change Authority to public in AuthenticationResult
- Uncomment out Sovereign Cloud test

* Update Sovereign cloud test to include actual json response

* adding header info for sovereign cloud test

* Add test for making instance discovery all because the authority is not in the instance cache

* Update unit test

* Addressing appveyor failure

* Change sovereign cloud test namespace to Ingegration

* Address appveyor issue

* Update to sovereign cloud tests based on authority migration changes

* Add default authority for blackforest in test constants

* Address PR comments
- change summary wording in AuthResult
- Fix formatting
- Simplify unit test

* fixing unit test from PR feedback

* Update mock to contain actual sovereign cloud authority

* Use ADAL code to call InstanceDiscovery

* Update HttpMessageHandlerFactory to notify when all mocks have been used for a test to ensure network calls are not made during testing
Add check for number of tokens in cache for tests that don't have it
Change ClearMockHandlers to InitializeMockProvider()
Add additional mocks where needed

* handling merge conflicts

* address merge conflict

* addressing PR comments and resharper edits

* addressing PR comments - fixing unit test

* Adding URL for mocks

* add const for variable

Commit:
ADAL.Net does no longer build with VS 2017 (Version 15.4.4)  (#896)

* fix build

* fix test project structure

* fix bad method call

SomkaPe committed:
adding PiiLoggingEnabled flag, refactoring (#877)

* adding PiiLoggingEnabled flag, refactoring

* adding delegate LogCallback with Pii option, making IAdalLogCallback obsolete

* fixing docs

* switching to piiApi, introducing piiLessToString for exceptions

* fixing tests

* refactoring

* adding NSubstitute dependency

* switching to double loggining pattern (separate log statements for non and pii logs)

* minor fix

* adding Header to Extensions

* fixing AdalClaimsChallengeException

CyberNyanta committed:
Fix ArgumentException message (#897)

jmprieur committed:
Updating the contribution guide (contributing.md) (#900)

* Updating the contribution guide (contributing.md) how to clone/fork to avoid long names issues, and how to restore the nuget package

* Adressing PR comments
+ changing the titles to use the "modern voice"

* Fixing a typo and explaining how to run the integration tests

Commit:
Fix build issues (#901)

* Update appveyor.yml

* run InstallAndroidSDK.ps1 [skip ci]

* Create InstallAndroidSDK.ps1

* Update appveyor.yml

jennyf19 committed:
Use CultureInvariant in Regex calls (#899)

* Use CultureInvariant in Regex calls
Use a static variable

* Addressing PR comments.

* removing underscore from updatedFromTemplate variable
#859

jennyf19 committed:
Jennyf/regex (#908)

* Use CultureInvariant in Regex calls
Use a static variable

* Addressing PR comments.

* removing underscore from updatedFromTemplate variable

* Addressing build issues
#859

jennyf19 committed:
Jennyf/nuget (#910)

* Use CultureInvariant in Regex calls
Use a static variable

* Addressing PR comments.

* removing underscore from updatedFromTemplate variable

* Addressing build issues

* fix nuget.org information
#906 #907

jennyf19 committed:
Adding functionality to print adal logs and to print adal logs with pii in the desktop app. (#912)

#877

jennyf19 committed:
Jennyf/desktop app pii (#914)

* Adding functionality to print adal logs and to print adal logs with pii in the desktop app.

* Even though the default is false, setting the LoggerCallbackHandler.PiiLoggingEnabled to false in case user continues using the app, and sets the pii logging to true, but acquire token without pii logs later.
#877

jmprieur committed:
Fixing #918  (tests don't show-up in the Test Explorer in VS2017.4+) (#919)

by upgrading the NuGet packages for the test adapter and test framework.
See minimum version requirements (https://docs.microsoft.com/en-us/visualstudio/test/live-unit-testing-faq#what-test-frameworks-does-live-unit-testing-support-and-what-are-the-minimum-supported-versions)

jennyf19 committed:
WIP

jennyf19 committed:
clean up sln file

jennyf19 committed:
add change log for release and update version to 3.17.4

jennyf19 committed:
Remove x5c claim comment from change log

jennyf19 committed:
adding in issue numbers to change log

henrik-me commented:
looks great. codeflow only handles 64 commits... I heard Jean-Marc is testing this one :)

jmprieur commented on src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationResult.cs:
```diff
@@ -122,9 +122,9 @@ internal AuthenticationResult(string accessTokenType, string accessToken, DateTi
-        /// Gets the authority which has issued tokens.
+        /// Gets the authority that has issued the token.
         /// </summary>
-        internal string Authority { get; set; }
```
This is an addition to the public API. We need to increase the minor number of the release (3.18.0, not 3.17.4)

jennyf19 replied:
I increased the minor number to 3.18.0

jmprieur commented on src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs:
```diff
@@ -57,8 +58,20 @@ public enum LogLevel
+    /// <param name="message">Pre-formatted log message</param>
+    /// <param name="containsPii">Indicates if the log message contains PII. If Logger.PiiLoggingEnabled is set to 
+    /// false then this value is always false.</param>
+    public delegate void LogCallback(LogLevel level, string message, bool containsPii);
```
This is an addition to the public API (and therefore we need to increase the release minor number) => 3.18 and not 3.17.4

jennyf19 replied:
done.

jmprieur commented on src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs:
```diff
@@ -76,18 +89,28 @@ public sealed class LoggerCallbackHandler
+        /// Flag to enable/disable logging of PII data. PII logs are never written to default outputs like Console, Logcat or NSLog.
+        /// Default is set to false.
+        /// </summary>
+        public static bool PiiLoggingEnabled { get; set; } = false;
```
This is an addition to the public API (and therefore we need to increase the release minor number) => 3.18 and not 3.17.4

jennyf19 replied:
done.

jmprieur reviewed (changes requested):
As explained in the details, there is a breaking change in the public API, which we can fix (as this is a change of namspace)

There are also, as expected additional changes to the public API, and therefore we need to increase the release number (3.18 and not 3.17.4)

The rest looks good to be, althought with that many files I might have missed issues (but for sure not in the public API :))

jmprieur committed:
WebBrowserNavigateErrorEventHandler was in Microsoft.IdentityModel.Clients.ActiveDirectory.Internal in ADAL.Net 3.17.3, and therefore changing the namespace of all this file to Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Platform constituted a public API breaking change (even if the delegate was not used by customers) (#953)

Restoring the namespace for this type only (WebBrowserNavigateErrorEventHandler)

jmprieur commented on changelog.txt:
```diff
@@ -1,3 +1,10 @@
+Version 3.17.4
```
Should be 3.18, really as we have additions to the public API.

jennyf19 replied:
done.

jmprieur commented:
We need to update the README to explain the new Logging mechanism. Proposed PR #955 for this.

jmprieur committed:
Updating the README.md to explain how to use the new Logging mechanism controlling if PII is viewed or not (from ADAL 3.18) (#955)

jennyf19 committed:
Updating version number to 3.18 for release in AssemblyInfo and changelog.txt. Also making 2 edits to ReadMe.md

jennyf19 commented on src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/CustomWebBrowser.cs:
```diff
@@ -30,7 +30,7 @@
 using System.Runtime.InteropServices;
 using System.Windows.Forms;
 
-namespace Microsoft.IdentityModel.Clients.ActiveDirectory.Internal
```
Got it. Thanks
